cyvlfeat/hog/hog.py

```diff
@@ -56,16 +56,26 @@ def hog(image, cell_size, variant='UoCTTI', n_orientations=9,
         only used for the 'DalalTriggs' variant.
     verbose : `bool`, optional
         If ``True``, be verbose.
+    visualize : `bool`, optional
+        Also return an image of the HOG. For each cell and orientation bin,
+        the image contains a line segment that is centered at the cell center,
+        is perpendicular to the midpoint of the range of angles spanned by the
+        orientation bin, and has intensity proportional to the corresponding
+        histogram value.
 
     Returns
     -------
-    output : `(R, R, D)` `float32` `ndarray`
-        ``R`` is is approximately the number of columns of ``image`` divided
-        by ``cell_size``. ``D`` is the number of feature dimensions.
+    output : `(RH, RW, D)` `float32` `ndarray`
+        ``RH`` and ``RW`` is approximately ``H / cellsize`` and ``W / cellsize``. ``D`` is the number of feature dimensions.
+    hog_image : 2D `uint8` `ndarray`, optional
+        A visualisation of the HOG image. Only provided if `visualize` is `True`.
     """
+    # vlfeat requires image in channel-first style
+    if image.ndim == 3:
```

Comment:
The only other thing I would request is a flag to be able to control this behaviour - in case the user wants to pass something where the channels are already first. Maybe something like `channels_first=False`?